mantidproject/mantid
# Set DetID before Q to prevent ray tracing when loading a peaks workspace

This changes how LoadNexusProcessed creates a `Peak` object to set the detector ID before Q. This means that the peaks workspace will not perform ray tracing and will just use the data loaded from file.

**To test:**
There is a peaks workspace attached to the issue that you can use for testing. This is now also in the unit test data.

 - Try opening the file before applying changes. 
 - Go get a coffee. On my machine this took ~5 mins.
 - Try opening the file with the changes applied. It should take significantly less time.

Fixes #19522 .

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
